server/src/service/OfferService.hh

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <oatpp/core/macro/component.hpp>
+#include <oatpp/web/protocol/http/Http.hpp>
+
+#include "database/MainDatabase.hh"
+#include "dto/PlaceDto.hh"
+#include "dto/StatusDto.hh"
+
+using Status = oatpp::web::protocol::http::Status;
+
+namespace server::service {
+
+    class OfferService {
+
+    private:
+        OATPP_COMPONENT(std::shared_ptr<database::MainDatabase>, database_);
+
+    public:
+        static std::shared_ptr<OfferService>
+        createShared() {
+            return std::make_shared<OfferService>();
+        }
+
+        oatpp::Object<dto::OfferDto>
+        createOne(
+                oatpp::Object<dto::OfferDto> const &dto
+        );
+
+        oatpp::Object<dto::OfferDto>
+        getOne(
+                oatpp::UInt64 const &id
+        );
+
+        oatpp::Object<dto::OfferPageDto>
+        search(
+                oatpp::String const &query,
+                oatpp::UInt64 const &limit = 20UL,
+                oatpp::UInt64 const &offset = 0UL
```

Comment:
`unsigned long` is not always 64 bits.
That is the main reason why we need CI and static analysis on Windows and MSVC.

Reply:
so what should we do?

Reply:
We can either leave it as `oatpp::UInt64 const &offset = 0`, or use `std::uint64_t{0}`.
The first option is probably better.

The issue with leaving it as `20UL` is that if we try to compile it with MSVC on Windows, it will fail because of conversion error. (unsigned long on Windows is 32bit not 64bit)

Reply:
Can we leave it as `0U`?

Reply:
nvm, it won't compile either way:
```
error: conversion from ‘int’ to ‘const oatpp::data::mapping::type::UInt64’ {aka ‘const oatpp::data::mapping::type::Primitive<long unsigned int, oatpp::data::mapping::type::__class::UInt64>’} is ambiguous
   38 |            UInt64 const& offset = 0);
```

Reply:
I tried some things, but:
```c++
UInt64 const& limit = std::uint64_t{20},
UInt64 const& offset = std::uint64_t{0});
```
is the only thing that worked.

Reply:
Now the funny thing is that the only reason why `UInt64 const& offset = 0` doesn't work is, because `0` can be both `uint64_t` and `nullptr_t`. That's why it's ambiguous.

Reply:
I forgor to mention
https://github.com/oatpp/oatpp/issues/449

Reply:
I'll regex replace it for now